R/internal.R

```diff
@@ -2,3 +2,93 @@
     # works like message() but prepends a timestamp
     message(date(), ": ", ...)
 }
+
+.rgbaConversion <- function(color, alpha = 0.5){
+    rgbastr <- NULL
+    if (!is.character(color)) {
+        rgbastr <- "invalid value"
+    }
+    tryCatch({
+        rgbaVal <- paste(c(col2rgb(color), alpha), collapse = ",")
+        rgbastr <- paste0("rgba(", rgbaVal, ")")
+    }, error = function(e) {
+        warning(paste("Ïnvalid color specified",color))
+    })
+    if (is.null(rgbastr)) {
+        rgbastr <- "invalid value"
+    }
+    return(rgbastr)
+}
+
+.validate_colors <- function(colors) {
+    valid_colors <- list()
+    for (color in colors) {
+        if (.rgbaConversion(color) != "invalid value") {
+           valid_colors <- append(valid_colors,color)
+        }
+    }
+    return(valid_colors)
```

Comment:
no need for explicit return call

Reply:
is there a specific reason .. isnt it more like a preference